src/theory/bv/theory_bv_solver.h

```diff
@@ -0,0 +1,126 @@
+/*********************                                                        */
+/*! \file theory_bv_solver.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Mathias Preiner
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Bit-vector solver interface.
+ **
+ ** Describes the interface for the internal bit-vector solver of TheoryBV.
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef CVC4__THEORY__BV__THEORY_BV_SOLVER_H
+#define CVC4__THEORY__BV__THEORY_BV_SOLVER_H
+
+#include "theory/bv/theory_bv.h"
+#include "theory/theory.h"
+
+namespace CVC4 {
+namespace theory {
+namespace bv {
+
+class TheoryBVSolver
```

Comment:
Also, another comment here, how about the name `BVSolver`?  This makes it clear that this isn't a Theory object.

Reply:
That's a good point, I'll do more renaming to make this clearer.